src/util/misc.ts

```diff
@@ -195,56 +233,145 @@ import { cos } from './index.ts';
+      };
+
       points.forEach(function (p, index) {
         var A = new fabric.Point(p.x, p.y), B, C;
         if (index === 0) {
           C = points[index + 1];
-          B = openPath ? getStrokeHatVector(fabric.util.createVector(C, A)).addEquals(A) : points[points.length - 1];
+          B = openPath ? A : points[points.length - 1];
         }
         else if (index === points.length - 1) {
           B = points[index - 1];
-          C = openPath ? getStrokeHatVector(fabric.util.createVector(B, A)).addEquals(A) : points[0];
+          C = openPath ? A : points[0];
         }
         else {
           B = points[index - 1];
           C = points[index + 1];
         }
-        var bisector = fabric.util.getBisector(A, B, C),
-            bisectorVector = bisector.vector,
+        B = new fabric.Point(B.x, B.y);
+        C = new fabric.Point(C.x, C.y);
+
+        if (openPath && (index === 0 || index === points.length - 1)) {
+          var D = index === 0 ? C : B,
+              scaledA = A.multiply(scale),
+              scaledD = D.multiply(scale);
+
+          var vector = fabric.util.createVector(
+                options.strokeUniform ? scaledA : A,
+                options.strokeUniform ? scaledD : D
+              ),
+              hatOrthogonalVector = fabric.util.getOrthogonalUnitVector(vector),
+              orthogonalVector = scaleHatVector(hatOrthogonalVector, s);
+
+          var proj1 = A.add(orthogonalVector),
+              proj2 = A.subtract(orthogonalVector);
+
+          coords.push(traceProjections ? storeTraceProjections(proj1, A, bisector) : proj1);
+          coords.push(traceProjections ? storeTraceProjections(proj2, A, bisector) : proj2);
+
+          return;
+        }
+
+        var bisector,
+            scaledA,
+            scaledB,
+            scaledC;
+        if (options.strokeUniform) {
+          scaledA = A.multiply(scale);
+          scaledB = B.multiply(scale);
+          scaledC = C.multiply(scale);
+          bisector = fabric.util.getBisector(scaledA, scaledB, scaledC);
+        }
+        else {
+          bisector = fabric.util.getBisector(A, B, C);
+        }
+
+        var bisectorVector = bisector.vector,
             alpha = bisector.angle,
             scalar,
             miterVector;
+
         if (options.strokeLineJoin === 'miter') {
           scalar = -s / Math.sin(alpha / 2);
-          miterVector = new fabric.Point(
-            bisectorVector.x * scalar * strokeUniformScalar.x,
-            bisectorVector.y * scalar * strokeUniformScalar.y
-          );
-          if (Math.hypot(miterVector.x, miterVector.y) / s <= options.strokeMiterLimit) {
-            coords.push(A.add(miterVector));
-            coords.push(A.subtract(miterVector));
+          miterVector = scaleHatVector(bisectorVector, scalar);
+
+          var strokeMiterLimit;
+          if (options.strokeUniform) {
+            var miterLimitVector = scaleHatVector(bisectorVector, options.strokeMiterLimit * s);
```

Comment:
this is tricky logic as well
do you want to explain it so we discuss and see we understand and agree on it?

Reply:
This part is kind of weird. It seems to me that the canvas recalculates the strokeMiterLimit when scaled with uniform stroke to maintain a proportion with the value initially set. 

When I kept the same strokeMiterLimit it didn't work.